src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java

```diff
@@ -150,6 +153,38 @@ void buildHearingDetails() {
         assertNotNull(hearingDetails.getPanelRequirements());
     }
 
+    @DisplayName("When isAdjournmentFlagEnabled is true, AdjournHearingsWindowMapping is used "
+        + "to calculate HearingWindow instead of HearingsWindowMapping")
+    @Test
+    void buildHearingDetailsAdjourned() {
+        SscsCaseData caseData = SscsCaseData.builder().adjournCaseGenerateNotice("Yes").build();
+        HearingWrapper wrapper = HearingWrapper.builder().caseData(caseData).build();
+
+        given(referenceDataServiceHolder.isAdjournmentFlagEnabled()).willReturn(true);
+
+        // AdjournHearingsWindow throws exception due to missing data
+        assertThatThrownBy(() -> HearingsDetailsMapping.buildHearingDetails(wrapper, referenceDataServiceHolder))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessageContaining("Unexpected adjournCaseNextHearingDateType: null for case id");
+
+    }
+
+    @DisplayName("When adjournCaseGenerateNotice is not null, isHearingAdjourned returns true")
+    @Test
+    void testIsHearingAdjourned() {
```

Comment:
More verbose / explanatory naming of test methods for readability and easier differentiation if future tests are required, etc.

Reply:
done